source/constructs/api/data_source/service.py

```diff
@@ -2470,10 +2482,24 @@
         raise BizException(MessageEnum.BIZ_UNKNOWN_ERR.get_code(),
                            MessageEnum.BIZ_UNKNOWN_ERR.get_msg())
 
+def query_account_network_agent(ec2_client: any):
+    res = []
+    vpc_list = [vpc['VpcId'] for vpc in ec2_client.describe_vpcs()['Vpcs']]
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `describe_vpcs` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `IsTruncated`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)